californium-core/src/test/java/org/eclipse/californium/rule/CoapNetworkRule.java

```diff
@@ -0,0 +1,144 @@
+import org.eclipse.californium.elements.util.DatagramFormatter;
+
+/**
+ * CoAP network rules for junit test using datagram sockets.
+ * 
+ * Manages also {@link NetworkConfig} and {@link EndpointManager}.
+ */
+public class CoapNetworkRule extends NetworkRule {
+
+	public static final Logger LOGGER = Logger.getLogger(CoapNetworkRule.class.getName());
+	private static final int DEFAULT_MESSAGE_THREADS = 1;
+	/**
+	 * CoAP datagram formatter. Used for logging.
+	 */
+	private static final DatagramFormatter FORMATTER = new DatagramFormatter() {
+
+		private DataParser parser = new UdpDataParser();
+
+		@Override
+		public String format(byte[] data) {
+			Message message = parser.parseMessage(data);
+			return message.toString();
+		}
+
+	};
+
+	/**
+	 * Number of message processing threads. Used to setup test
+	 * {@link NetworkConfig}.
+	 * 
+	 * @see #createTestConfig()
+	 * @see #createStandardTestConfig()
+	 * @see #getStandardTestConfig()
+	 */
+	private final AtomicInteger messageThreads = new AtomicInteger(DEFAULT_MESSAGE_THREADS);
+
+	/**
+	 * Create rule supporting provided modes.
+	 * 
+	 * @param modes supported datagram socket implementation modes.
+	 */
+	public CoapNetworkRule(Mode... modes) {
+		super(FORMATTER, modes);
+		this.messageThreads.set(DEFAULT_MESSAGE_THREADS);
+	}
+
+	/**
+	 * Set number of message processing threads. Using multiple threads for
+	 * sending and receiving may result in reordering of messages. Therefore the
+	 * default is {@link #DEFAULT_MESSAGE_THREADS} to ensure, that no such
+	 * reorder happens.
+	 * 
+	 * @param threads number of threads.
+	 * @return this rule
+	 */
+	public CoapNetworkRule setMessageThreads(int threads) {
+		if (1 > threads) {
+			throw new IllegalArgumentException("number of message threads must be at least 1, not " + threads + "!");
+		}
+		messageThreads.set(threads);
+		return this;
+	}
+
+	@Override
+	protected void initNetwork(boolean first) {
+		if (first) {
+			EndpointManager.reset();
+		}
+		createStandardTestConfig();
+		super.initNetwork(first);
+	}
+
+	@Override
+	protected void closeNetwork() {
+		super.closeNetwork();
+		EndpointManager.reset();
+		messageThreads.set(DEFAULT_MESSAGE_THREADS);
+		NetworkConfig.setStandard(null);
+	}
+
+	/**
+	 * Create new standard network configuration for testing.
+	 * 
+	 * @return fresh standard network configurations. Changes are visible to
+	 *         other usage of the standard configuration.
+	 * @see NetworkConfig#getStandard()
+	 */
+	public NetworkConfig createStandardTestConfig() {
+		NetworkConfig config = createTestConfig();
+		NetworkConfig.setStandard(config);
+		return config;
+	}
+
+	/**
+	 * Get standard network configuration for testing.
+	 * 
+	 * @return standard network configurations. Changes are visible to other
+	 *         usage of the standard configuration.
+	 * @see NetworkConfig#getStandard()
+	 */
+	public NetworkConfig getStandardTestConfig() {
+		return NetworkConfig.getStandard();
```

Comment:
The intention is, that this `Rule` prepares the `NetworkConfig.getStandard()` within its scope according the setup of this `Rule` without using a file (see `createStandardTestConfig`). But yes, if used wrong, it may not work as intended. I think adjusting the documentation and make the implementation more safe against miss use should be possible.